src/test/cpp/gtester/HTTPProtoTest.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <gtest/gtest.h>
+#include <thread>
+#include <iterator>
+
+#include "http/prototype.h"
+
+#include "Poco/Net/HTTPClientSession.h"
+#include "Poco/Net/NetException.h"
+#include "Poco/Net/HTTPRequest.h"
+#include "Poco/Net/HTTPResponse.h"
+#include <Poco/Net/HTTPCredentials.h>
+#include "Poco/StreamCopier.h"
+#include "Poco/NullStream.h"
+#include "Poco/Path.h"
+#include "Poco/URI.h"
+#include "Poco/Exception.h"
+
+using namespace std;
+using namespace Stride;
+
+// Poco
+using Poco::Net::HTTPClientSession;
+using Poco::Net::ConnectionRefusedException;
+using Poco::Net::HTTPRequest;
+using Poco::Net::HTTPResponse;
+using Poco::Net::HTTPMessage;
+using Poco::StreamCopier;
+using Poco::Path;
+using Poco::URI;
+using Poco::Exception;
+
+
+void runServer(VizProto* p){
+    cout << "Running server!" << endl;
+    p->run();
+}
+
+string doRequest(Poco::Net::HTTPClientSession& session, Poco::Net::HTTPRequest& request, Poco::Net::HTTPResponse& response){
+    session.sendRequest(request);
+    istream& rs = session.receiveResponse(response);
+    
+    if (response.getStatus() != Poco::Net::HTTPResponse::HTTP_UNAUTHORIZED){
+        // return the answer as a string
+        return string(istreambuf_iterator<char>(rs), {});
+    }else{
+        //it went wrong...?
+        return string("");
```

Comment:
Using the empty string as a failure state is pretty icky. It looks like this should never happen, so throwing an exception seems more reasonable.

(Also, I’d invert the `if` like so:)

```cpp
if (response.getStatus() == Poco::Net::HTTPResponse::HTTP_UNAUTHORIZED) {
  // Something went wrong.
  throw some_kind_of_error("Poco HTTP request got unauthorized response!");
}

return string(istreambuf_iterator<char>(rs), {});
```